java-components/cache/src/main/java/com/redhat/hacbs/artifactcache/util/RequestCleanup.java

```diff
@@ -0,0 +1,45 @@
+package com.redhat.hacbs.artifactcache.util;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.annotation.PreDestroy;
+import javax.enterprise.context.RequestScoped;
+
+import io.quarkus.arc.Arc;
+import io.quarkus.arc.Unremovable;
+import io.quarkus.logging.Log;
+
+/**
+ * A utility to ensure resources are correctly cleaned up at the end of the request.
+ * <p>
+ * This is useful as some HTTP client resources have lifecycles that are not super well defined so
+ * a try/finally block cannot be used. This ensures that these resources are correctly cleaned up.
```

Comment:
interesting .... fwiw I at least have not seen the analogous situation pop up with the golang k8s clients (which are just http clients under the covers).  Wonder if that is a difference in the clients, or my lack of exposure to intermittent problems out there.

Reply:
This was actually an issue with the HTTP Maven client, not the golang one. I think the leak is actually me just missing a close, I will update the 'async maven-metadata.xml' PR to deal with it properly and remove this.

Basically you can't just use these resources in a try/finally block, as the results of the HTTP connection may have to 'bubble up (in some cases being streamed straight back to the client if we decide not to cache)'.

Reply:
ok thanks for the confirmation, as I suspected, that this was specific to the client you were using / how you were using it